modules/wsl-distro.nix

```diff
@@ -211,15 +206,20 @@ in
         (mkIf cfg.nativeSystemd {
           wsl.wslConf = {
             user.default = config.users.users.${cfg.defaultUser}.name;
-            boot.systemd = true;
+            boot = {
+              systemd = true;
+              initPath = "${nativeUtils}/bin/systemd-shim";
+              initShutdownCommand = pkgs.writeShellScript "nixos-wsl-init-shutdown" ''
+                ${pkgs.systemd}/bin/systemctl reboot
```

Comment:
```suggestion
                ${pkgs.systemd}/bin/systemctl reboot "$@"
```
just incase

Reply:
I'm actually pretty sure we can avoid the shell script here and just write the string directly.